crates/polars-time/src/round.rs

```diff
@@ -44,3 +44,43 @@ impl PolarsRound for DateChunked {
             .into_date())
     }
 }
+
+#[cfg(feature = "dtype-duration")]
+impl PolarsRound for DurationChunked {
+    fn round(&self, every: Duration, offset: Duration, _tz: Option<&Tz>) -> PolarsResult<Self> {
+        if every.negative {
+            polars_bail!(ComputeError: "cannot round a Duration to a negative duration")
+        }
+        if !every.is_constant_duration() {
+            polars_bail!(InvalidOperation: "Cannot round a Duration series to a non-constant duration.");
+        }
```

Comment:
been thinking about this one again, and I'm not sure the error message is clear enough

I'm addressing this as part of a separate PR, will update you when it's ready